src/main/kotlin/lotto/WinningMachine.kt

```diff
@@ -0,0 +1,51 @@
+package lotto
+
+class WinningMachine(winningString: String) {
+
+    val winningLotto: Lotto
+
+    init {
+        val stringNumbers = getStringNumbers(winningString)
+        winningLotto = Lotto(getWinningNumbers(stringNumbers))
+    }
+
+    private fun getStringNumbers(winningString: String): List<String> {
+        val stringNumbers = winningString.split(",")
+
+        require(stringNumbers.isNotEmpty()) { "input string delimiter" }
+        require(stringNumbers.size == LOTTO_NUMBERS_COUNT) { "input string delimiter count" }
+
+        return stringNumbers
+    }
+
+    private fun getWinningNumbers(stringNumbers: List<String>): List<LottoNumber> {
+        val lottoNumbers = stringNumbers.map { LottoNumber(it.trim()) }
+        require(lottoNumbers.size == lottoNumbers.distinct().size) { "input string numbers duplicates" }
+        return lottoNumbers
+    }
+
+    fun winningResult(lottoList: List<Lotto>): HashMap<RANKING, Int> {
+        val winningResult = HashMap<RANKING, Int>()
+
+        lottoList.forEach { lotto ->
+            val winningNumbers = winningLotto.numbers.filter { lotto.numbers.contains(it) }
```

Comment:
```suggestion
            val winningNumbers = winningLotto.numbers.filter { lotto.contains(it) }
```
프로퍼티를 꺼내지 않고 메시지를 던지는 구현은 어떻게 생각하시나요 ? 
앞으로 미션을 진행하며 data class 가 아닌 class의 프로퍼티의 가시성을 private으로 유지하고, 메시지(메소드)를 제공할수있도록 구현해보는것을 권장 드립니다.